### PR TITLE
mgr: use un-deprecated APIs to initialize Python interpretor

### PR DESCRIPTION
**Pull request by tchaikov**
`Py_SetProgramName()`, `PySys_SetArgv()` and `Py_GetPath()` were deprecated since CPython 3.11, see https://docs.python.org/3/c-api/init_config.html . `Py_InitializeFromConfig()` and friends were introduced by CPython 3.8,
fortunately we don't need to support CPython 3.6 which is shipped by CentOS8. so we just need to to be compatible with the Python 3.9 and up. please note that ubuntu:jammy ships Python 3.10, see https://packages.ubuntu.com/jammy/amd64/python3 .

in this changeset, we use the new machinary to initialize the Python interpreter. so that this piece of code can be tested on 

this change addresses following compiling warning:

```
[428/753] Building CXX object src/mgr/CMakeFiles/ceph-mgr.dir/PyModuleRegistry.cc.o
/var/ssd/ceph/src/mgr/PyModuleRegistry.cc: In member function ‘void PyModuleRegistry::init()’:
/var/ssd/ceph/src/mgr/PyModuleRegistry.cc:49:20: warning: ‘void Py_SetProgramName(const wchar_t*)’ is deprecated [-Wdeprecated-declarations]
   49 |   Py_SetProgramName(const_cast<wchar_t*>(WCHAR(MGR_PYTHON_EXECUTABLE)));
      |   ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/python3.12/Python.h:94,
                 from /var/ssd/ceph/src/mgr/PyModule.h:22,
                 from /var/ssd/ceph/src/mgr/PyModuleRegistry.h:18,
                 from /var/ssd/ceph/src/mgr/PyModuleRegistry.cc:14:
/usr/include/python3.12/pylifecycle.h:37:38: note: declared here
   37 | Py_DEPRECATED(3.11) PyAPI_FUNC(void) Py_SetProgramName(const wchar_t *);
      |                                      ^~~~~~~~~~~~~~~~~`
```

this change partially reverts 57e9375a40f192441e325969af4fc1087dd209bf. since we don't need to support Python 3.6 anymore, we don't need to worry about the issue addressed by #56617

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
cc @adk3798   Shall we run this thru the orch test suite?

**Comment by adk3798** [comment]
> cc @adk3798 Shall we run this thru the orch test suite?

tagged, will do a run

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-06-28_04:58:18-orch:cephadm-wip-adk-testing-2024-06-27-0902-distro-default-smithi/

Failures all fell under

- cluster log failures (known issue, being cleaned up)
- mds_upgrade_Sequence (known issue)
- staggered upgrade with agent on (known issue)

Nothing to block merging

**Review by adk3798 — APPROVED** [review]
This no longer seems to cause issues in orch runs now that we have centos 9 based containers. Do you think we need another review from the core team again or do you consider the approval on the previous PR enough @tchaikov ?

**Comment by tchaikov** [comment]
@adk3798 hi Adam, thank you for your review and test!

@ronen-fr hi Ronen, you reviewed #55436 which was reverted due to the incompatibility with Python 3.6. this change is a subset of #55436 -- i removed the part which intended to keep the backward compatibility with Python < 3.8.  are you fine with this changeset as well ? if the answer is "yes", i am going to merge it.

**Comment by tchaikov** [comment]
seems there are no objections. i am merging this changeset.

**Comment by rkhudov** [comment]
@tchaikov do we need to update default value for [python3_pkgversion](https://github.com/ceph/ceph/blob/main/ceph.spec.in#L141) in ceph.spec.in from 3 to 3.9?

**Pull request merged**
